src/main/scala/org/kframework/minikore/implementation/MiniKore.scala

```diff
@@ -1,122 +1,163 @@
+  }
 
-  case class Import(name: i.Name, att: Attributes) extends Sentence
+  case class Application(_1: p.Symbol, args: Seq[p.Pattern]) extends p.Application {
+    def build(_1: p.Symbol, args: Seq[p.Pattern]): Application = Application(_1, args)
+  }
 
-  case class SortDeclaration(sort: i.Sort, att: Attributes) extends Sentence
+  case class Top() extends p.Top {
+    def build(): Top = Top()
+  }
 
-  case class SymbolDeclaration(sort: i.Sort, symbol: i.Symbol, args: Seq[i.Sort], att: Attributes) extends Sentence
+  case class Bottom() extends p.Bottom {
+    def build(): Bottom = Bottom()
+  }
 
-  case class Rule(pattern: i.Pattern, att: Attributes) extends Sentence
+  case class And(_1: p.Pattern, _2: p.Pattern) extends p.And {
+    def build(_1: p.Pattern, _2: p.Pattern): And = And(_1, _2)
+  }
 
-  case class Axiom(pattern: i.Pattern, att: Attributes) extends Sentence
+  case class Or(_1: p.Pattern, _2: p.Pattern) extends p.Or {
+    def build(_1: p.Pattern, _2: p.Pattern): Or = Or(_1, _2)
+  }
 
-  case class Variable(_1: i.Name, _2: i.Sort) extends i.Variable {
-    def build(_1: i.Name, _2: i.Sort): Variable = Variable(_1, _2)
+  case class Not(_1: p.Pattern) extends p.Not {
+    def build(_1: p.Pattern): Not = Not(_1)
   }
 
-  case class DomainValue(_1: i.Symbol, _2: i.Value) extends i.DomainValue {
-    def build(_1: i.Symbol, _2: i.Value): DomainValue = DomainValue(_1, _2)
+  case class Implies(_1: p.Pattern, _2: p.Pattern) extends p.Implies {
+    def build(_1: p.Pattern, _2: p.Pattern): Implies = Implies(_1, _2)
   }
 
-  case class Application(_1: i.Symbol, args: Seq[i.Pattern]) extends i.Application {
-    def build(_1: i.Symbol, args: Seq[i.Pattern]): Application = Application(_1, args)
+  case class Exists(_1: p.Variable, _2: p.Pattern) extends p.Exists {
+    def build(_1: p.Variable, _2: p.Pattern): Exists = Exists(_1, _2)
   }
 
-  case class Top() extends i.Top {
-    def build(): Top = Top()
+  case class ForAll(_1: p.Variable, _2: p.Pattern) extends p.ForAll {
+    def build(_1: p.Variable, _2: p.Pattern): ForAll = ForAll(_1, _2)
   }
 
-  case class Bottom() extends i.Bottom {
-    def build(): Bottom = Bottom()
+  case class Next(_1: p.Pattern) extends p.Next {
+    def build(_1: p.Pattern): Next = Next(_1)
   }
 
-  case class And(_1: i.Pattern, _2: i.Pattern) extends i.And {
-    def build(_1: i.Pattern, _2: i.Pattern): And = And(_1, _2)
+  case class Rewrite(_1: p.Pattern, _2: p.Pattern) extends p.Rewrite {
+    def build(_1: p.Pattern, _2: p.Pattern): Rewrite = Rewrite(_1, _2)
   }
 
-  case class Or(_1: i.Pattern, _2: i.Pattern) extends i.Or {
-    def build(_1: i.Pattern, _2: i.Pattern): Or = Or(_1, _2)
+  case class Equals(_1: p.Pattern, _2: p.Pattern) extends p.Equals {
+    def build(_1: p.Pattern, _2: p.Pattern): Equals = Equals(_1, _2)
   }
 
-  case class Not(_1: i.Pattern) extends i.Not {
-    def build(_1: i.Pattern): Not = Not(_1)
+}
+
+
+/** Helpers for building MiniKore definitions **/
```

Comment:
This should be moved to a separate file, in my opinion. This object should be paramteric over a builder interface, that builds outer structures.